cluster-operator/src/main/java/io/strimzi/operator/cluster/model/ModelUtils.java

```diff
@@ -422,27 +413,77 @@ public static void jvmSystemProperties(List<EnvVar> envVars, JvmOptions jvmOptio
      * @param jvmOptions JVM options
      */
     public static void jvmPerformanceOptions(List<EnvVar> envVars, JvmOptions jvmOptions) {
-        StringBuilder jvmPerformanceOpts = new StringBuilder();
-
-        Map<String, String> xx = jvmOptions != null ? jvmOptions.getXx() : null;
-        if (xx != null) {
-            xx.forEach((k, v) -> {
-                jvmPerformanceOpts.append(' ').append("-XX:");
-
-                if ("true".equalsIgnoreCase(v))   {
-                    jvmPerformanceOpts.append("+").append(k);
-                } else if ("false".equalsIgnoreCase(v)) {
-                    jvmPerformanceOpts.append("-").append(k);
-                } else  {
-                    jvmPerformanceOpts.append(k).append("=").append(v);
-                }
-            });
+        parseJvmPerformanceOptions(jvmOptions)
+            .map(envVar -> ContainerUtils.createEnvVar(AbstractModel.ENV_VAR_KAFKA_JVM_PERFORMANCE_OPTS, envVar))
+            .ifPresent(envVars::add);
+    }
+
+    /**
+     * Parses JVM performance options from {@link JvmOptions#getXx()} property into a space-separated string of JVM flags.
+     * If performance options object is null or XX options are not configured then {@link Optional#empty()} will be returned.
+     *
+     * @param jvmOptions JVM options to parse
+     * @return optional comma-separated string of JVM flags
+     */
+    private static Optional<String> parseJvmPerformanceOptions(JvmOptions jvmOptions) {
```

Comment:
The changes in this class look good to me. But it is quite a lot of various methods. Do you think it would make sense to have some `JvmOptionUtils` class for them or something like that?